### PR TITLE
Improve error reporting for operator-courier related errors

### PR DESCRIPTION
**Pull request by chmeliik**
Addresses [#29](https://github.com/release-engineering/operators-manifests-push-service/issues/29).

### CODE REVIEW

File: docs/usage/v1.md [hunk 1]
```diff
@@ -44,14 +44,19 @@ Error messages have following format:
   "message": "<detailed error description>",
 }
 ```
+There may potentially be other fields providing further details,
```
Comment (MartinBasti) [inline docs/usage/v1.md]: I'd be more explicit which particular errors will provide that information. It can be part of explanation cell IMO

File: omps/errors.py [hunk 2]
```diff
@@ -13,6 +13,14 @@ class OMPSError(Exception):
     """Base OMPSError"""
     code = 500
 
+    def to_json(self):
```
Comment (MartinBasti) [inline omps/errors.py]: This is not a JSON, it returns python dict, this method should be called differently: `to_dict` for example.

File: tests/test_quay.py [hunk 3]
```diff
@@ -124,8 +236,8 @@ def test_push_operator_manifest_publish_repo(self, mocked_op_courier_push):
          .once())
         qo.push_operator_manifest(self.repo, self.version, self.source_dir)
 
-    def test_push_operator_manifest_publish_repo_no_public(
-            self, mocked_op_courier_push):
+    @pytest.mark.usefixtures('mocked_op_courier_push')
+    def test_push_operator_manifest_publish_repo_no_public(self):
```
Comment (MartinBasti) [inline tests/test_quay.py]: How is this change related to this PR? ;)

File: omps/errors.py [hunk 4]
```diff
@@ -28,6 +36,58 @@ class QuayCourierError(OMPSError):
...
+        :param validation_info: errors and warnings found during validation,
+                                if available
+        """
+        super().__init__(msg)
+        self.validation_info = validation_info
+
+    def to_json(self):
+        data = super().to_json()
+        if self.validation_info is not None:
+            data.update({'validation_info': self.validation_info})
```
Comment (MartinBasti) [inline omps/errors.py]: IMO we should always return that key in response, but with empty dict

File: omps/errors.py [hunk 5]
```diff
@@ -28,6 +36,58 @@ class QuayCourierError(OMPSError):
...
+        :param msg: the message this exception should have
+        :param quay_response: Quay error response json, if available
+        """
+        super().__init__(msg)
+        self.quay_response = quay_response
+
+    def to_json(self):
+        data = super().to_json()
+        if self.quay_response is not None:
+            data.update({'quay_response': self.quay_response})
```
Comment (MartinBasti) [inline omps/errors.py]: IMO we should always return that key in response, but with empty dict when undefined

File: tests/test_quay.py (hunk 3 above)
Comment (chmeliik) [reply tests/test_quay.py]: It just kind of happened as I was going through tests, do you want it changed back? :D

Comment (MartinBasti) [reply tests/test_quay.py]: Not this time, I forgive you for now. But next time unrelated changes should be in separate PR

**Comment by MartinBasti** [comment]
Please set requirement for operator-courier >= 1.2.1 in setup.py

**Comment by chmeliik** [comment]
Made the changes as @MartinBasti requested.

**Comment by chmeliik** [comment]
Rebased and fixed issues.
